x/concentrated-liquidity/model/codec.go

```diff
@@ -14,6 +16,12 @@ func RegisterCodec(cdc *codec.LegacyAmino) {
 }
 
 func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+	registry.RegisterInterface(
+		"osmosis.concentratedliquidity.v1beta1.PoolI",
```

Comment:
I might be missing context, why not register as swapRouter PoolI as is?

Reply:
Not quite sure what you mean? All I know is that PoolI was not registered in this module, so when we returned PoolI from pool queries, we were not able to unmarshal the response

Reply:
It think we should make this consistent with:
https://github.com/osmosis-labs/osmosis/blob/a1c13a6ebd92fdb46f2673f1702f5a430264964e/x/gamm/pool-models/balancer/codec.go#L24

Reply:
Addressed